packages/circuit-ui/components/FieldAtoms/FieldWrapper.tsx

```diff
@@ -13,25 +13,47 @@
  * limitations under the License.
  */
 
-import { HTMLAttributes } from 'react';
-import isPropValid from '@emotion/is-prop-valid';
+import { forwardRef, HTMLAttributes, Ref } from 'react';
+import { css } from '@emotion/react';
 
 import styled from '../../styles/styled';
-import { disableVisually } from '../../styles/style-mixins';
+import { EmotionAsPropType } from '../../types/prop-types';
+
+import { CLASS_DISABLED } from './constants';
 
 export interface FieldWrapperProps extends HTMLAttributes<HTMLDivElement> {
+  as?: EmotionAsPropType;
   /**
    * Trigger disabled styles on the component.
    */
   disabled?: boolean;
 }
 
-const disabledStyles = ({ disabled }: FieldWrapperProps) =>
-  disabled && disableVisually();
+const wrapperStyles = () =>
+  css`
+    &.${CLASS_DISABLED} {
+      pointer-events: none;
+    }
+  `;
+
+const Wrapper = styled('div')(wrapperStyles);
 
 /**
  * @private
  */
-export const FieldWrapper = styled('div', {
-  shouldForwardProp: (prop) => isPropValid(prop) && prop !== 'disabled',
-})<FieldWrapperProps>(disabledStyles);
+export const FieldWrapper = forwardRef(
+  (
+    { children, disabled, className = '', ...props }: FieldWrapperProps,
+    ref: Ref<HTMLDivElement>,
+  ) => (
+    <Wrapper
+      ref={ref}
+      className={disabled ? `${className} ${CLASS_DISABLED}` : className}
```

Comment:
Previously, the disabled state was applied to all atoms using `opacity: 0.5`. Now, we override the color of each atom individually. To avoid having to pass the `disabled` prop to each atom, I decided to add a custom classname to the `FieldWrapper` and use the cascade to apply disabled styles to the other atoms. WDYT?